packages/dynamoose/lib/General.ts

```diff
@@ -14,7 +14,9 @@ export type KeyObject = {[attribute: string]: string | number};
 // An item representing a DynamoDB key
 export type InputKey = string | number | KeyObject;
 
-interface ModelItemConstructor<T extends Item> {
+type Expand<T> = {[KeyType in keyof T]: T[KeyType]} & {};
```

Comment:
Why the `& {}`?

Also, can you give an example of what this would look like? For some reason this type seems very complex at first glance. It's not obviously clear what it's doing and what it's purpose is. Maybe add a comment above with an example or something? Or a better description?

Reply:
Also should this be in a separate file?

Reply:
Good points! This type definitely could use a JSDoc explaining its function & intended use. Also agree that it should be moved to a separate file. Best location for it from what I can see would be `lib/Types.ts`. How does that sound?

Reply:
`lib/Types.ts` works for me 🎉